trigger/interface.go

```diff
@@ -14,10 +14,10 @@ type eventHandler interface {
 	// EventLogEntry
 	Fetch(*event.EventLogEntry) error
 
-	// Selectors creates a slice of selectors suitable for fetching
-	// subscriptions for the event. Selectors should not perform
-	// any fetch operations.
-	Selectors() []event.Selector
+	// Selectors creates a map of selectors to the event's attributes
+	// suitable for fetching subscriptions for the event.
+	// Selectors should not perform any fetch operations.
+	Selectors() map[string][]string
```

Comment:
renamed to `Attributes`
I struggled with this in the initial implementation and then forgot about it. Lmk what you think about the new implementation. Reflection which could have been less verbose, but was avoided because it's complicated, slow, and, in this case, unnecessary.